Library/Homebrew/utils.sh

```diff
@@ -0,0 +1,47 @@
+puts_with_style() {
+  local message output_fd="$1" plain="$2" styled="$3"
+  shift 3
+
+  if [[ $# -eq 0 ]]
+  then
+    IFS= read -d '' -r message # Read message from `stdin`.
+    message="${message%$'\n'}" # Trim trailing newline.
+  else
+    message="$*"
+  fi
+
+  if [[ -t "$output_fd" ]] # If FD is a TTY, style message with color/bold/...
+  then
+    # shellcheck disable=SC2059
+    printf "$styled\n" "$message"
+  else
+    # shellcheck disable=SC2059
+    printf "$plain\n" "$message"
+  fi
+}
+
+ohai() {
+  # Use bold and blue color if `stdout` is a TTY.
+  puts_with_style 1 "==> %s" "\033[1;34m==>\033[1;39m %s\033[0m" "$@"
```

Comment:
So what do I do? The best I was able to do in `puts_with_style` was to immediately assign the first argument to a local variable named `output_fd` and to put this function just above the others and well within sight …

One could imagine an interface like `puts_with_style --fd=1 --plain-format="==> %s" --tty-format="\033[1;34m==>\033[1;39m %s\033[0m" -- <message>` but that feels a bit overengineered for a simple function that is only used internally to reduce code duplication among `odie` and friends.